Plugins/Creature/Creature.cpp

```diff
@@ -2865,3 +2866,84 @@ NWNX_EXPORT ArgumentStack SetPreferredAttackDistance(ArgumentStack&& args)
 
     return {};
 }
+
+static void InitEffectImmunityBypassHook()
+{
+    static Hooks::Hook pGetEffectImmunity_hook =
+        Hooks::HookFunction(Functions::_ZN17CNWSCreatureStats17GetEffectImmunityEhP12CNWSCreaturei,
+        (void*)+[](CNWSCreatureStats *pThis, uint8_t nType, CNWSCreature * pVersus, int32_t bConsiderFeats = true) -> int32_t
+        {
+            int32_t BypassCounter = 0;
+            if (auto IncomingBypass = pThis->m_pBaseCreature->nwnxGet<int32_t>("BYPASS_EFF_IMM_IN" + std::to_string(nType)))
+            {
+                if (rand() % 100 < abs(IncomingBypass.value()))
+                {
+                    if (IncomingBypass.value() > 0) //Positive being "Force not immune"
+                        BypassCounter -= 1;
+                    else //Negative being "Force as immune"
+                        BypassCounter += 1;
+                }
+            }
+            if (auto OutgoingBypass = pVersus->nwnxGet<int32_t>("BYPASS_EFF_IMM_OUT" + std::to_string(nType)))
+            {
+                if (rand() % 100 < abs(OutgoingBypass.value()))
+                {
+                    if (OutgoingBypass.value() > 0) //Positive being "Force not immune"
+                        BypassCounter -= 1;
+                    else //Negative being "Force as immune"
+                        BypassCounter += 1;
+                }
+            }
+
+            if (BypassCounter > 0)
+                return true;
+            else if (BypassCounter < 0)
+                return false;
+            else
+                return pGetEffectImmunity_hook->CallOriginal<int32_t>(pThis, nType, pVersus, bConsiderFeats);
+        }, Hooks::Order::Late);
```

Comment:
You can just put this bit inside `NWNX_EXPORT ArgumentStack SetBypassEffectImmunity(ArgumentStack&& args)` at the top, no need for the bool/function.